File: src/main/java/com/web/fitmaster/model/Membership.java
package com.web.fitmaster.model;

import com.web.fitmaster.model.enums.MembershipStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "memberships")

public class Membership {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "membership_id")
    private Long membershipId;

    @ManyToOne
    @JoinColumn(name = "member_id")
    private User member;

    @ManyToOne
    @JoinColumn(name = "package_id")
    private Package pkg;

    @Enumerated(EnumType.STRING)
    private MembershipStatus status;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "end_date", nullable = false)
    private LocalDate endDate;

    private BigDecimal price;

    private BigDecimal debt;

    private String description;




}

File: frontend/src/types/membership.ts

export interface MembershipRequest {
  startDate: string;
  price: number;
  debt: number;
  packageId:string;
  description:string;
}

export interface Membership {
  id:string;
  memberId:string;
  startDate:string;
  endDate:string;
  price: number;
  debt: number;
  packageName:string;
  description:string;
  timestamp:string;
}


export interface MembershipHistory {
  id:string;
  price: number;
  debt: number;
  packageName:string;
  description:string;
  timestamp:string;
}